FileSystemSimulator/FileSystem/FileSystem.cpp

```diff
@@ -199,7 +199,7 @@ int FileSystem::read(size_t i_index, char* i_mem_area, size_t i_count)
 		if (local_count <= buffer_space) {
 			std::memcpy(i_mem_area + have_written, entry->buffer + buffer_pos, local_count);
 			entry->cur_pos += local_count;
-			return entry->cur_pos - 1;
+			return i_count;
```

Comment:
```suggestion
			return have_written;
```
i_count может быть больше чем fd.file_size - entry->cur_pos